main.nf

```diff
@@ -313,61 +313,60 @@ process alignWithBowtie {
     """
 }
 
+
 /*
- * Sort the bam file with samtools and index the result.
+ * Take the reads and process them to get the remapped variants
+ *
  */
-process sortBam {
+process readsToRemappedVariants {
 
     input:
         path "reads_aligned.bam" from reads_aligned_bam
+        path "genome.fa" from params.newgenome
 
     output:
-        path "reads_aligned.sorted.bam" into reads_sorted_bam
-        path "reads_aligned.sorted.bam.bai" into bam_index
+        path "variants_remapped.vcf" into variants_remapped
 
     """
-    samtools sort reads_aligned.bam -o reads_aligned.sorted.bam
-    samtools index reads_aligned.sorted.bam
+    # Ensure that we will use the reads_to_remapped_variants.py from this repo
+    ${baseDir}/variant_remapping_tools/reads_to_remapped_variants.py -i reads_aligned.bam \
+        -o variants_remapped.vcf -f $params.flankingseq \
+        -s $params.scorecutoff -d $params.diffcutoff --max_alignment 1 \
+        --newgenome genome.fa
     """
 }
 
 /*
- * Take the reads and process them to get the remapped variants
- *
- */
-process readsToRemappedVariants {
+ * Sort VCF file
+ */
+process sortVCF {
 
     input:
-        path "reads_aligned.sorted.bam" from reads_sorted_bam
-        path "genome.fa" from params.newgenome
+        path "variants_remapped.vcf" from variants_remapped
 
     output:
-        path "variants_remapped.vcf" into variants_remapped
+        path "variants_remapped_sorted.vcf" into variants_remapped_sorted
 
     """
-    # Ensure that we will use the reverse_strand.py from this repo
-    ${baseDir}/variant_remapping_tools/reads_to_remapped_variants.py -i reads_aligned.sorted.bam \
-        -o variants_remapped.vcf -f $params.flankingseq \
-        -s $params.scorecutoff -d $params.diffcutoff --newgenome genome.fa
+    cat variants_remapped.vcf | awk '\$1 ~ /^#/ {print \$0;next} {print \$0 | "sort -k1,1 -k2,2n"}' > variants_remapped_sorted.vcf
```

Comment:
Can we use `bcftools sort` here? Unless there is a reason not to
```suggestion
        bgzip variants_remapped.vcf
        tabix variants_remapped.vcf.gz
        bcftools sort -o variants_remapped_sorted.vcf -Ov variants_remapped.vcf.gz
```

Reply:
Not in this stage because the header has not been added back yet. but in subsequent stage it would work.